mule-apikit-common/src/main/java/org/mule/apikit/common/APISyncUtils.java

```diff
@@ -47,4 +47,66 @@ public static String toApiSyncResource(String s) {
                                       EXCHANGE_TYPE, resourceParts[length - 1]);
     return apiSyncResource;
   }
+
+  public static boolean compareResources(String resourceA, String resourceB, Boolean checkVersion) {
+    APISyncResource apiSyncResourceA = null;
+    APISyncResource apiSyncResourceB = null;
+    try {
+      apiSyncResourceA = new APISyncResource(resourceA);
+      apiSyncResourceB = new APISyncResource(resourceB);
+    } catch (APISyncResourceException e) {
+      return false;
+    }
+
+    return apiSyncResourceA.equals(apiSyncResourceB, checkVersion);
+  }
+
+  private static class APISyncResource {
+
+    private String groupId;
+    private String artifact;
+    private String version;
+    private String classifier;
+    private String packager;
+    private String file;
+
+
+    APISyncResource(String resource) throws APISyncResourceException {
+      if (!APISyncUtils.isSyncProtocol(resource))
+        throw new APISyncResourceException("Invalid APISync Resource");
+
+      String[] parts = resource.substring(APISyncUtils.API_SYNC_PROTOCOL.length()).split(":");
+
+      if (parts.length != 6)
+        throw new APISyncResourceException("Invalid APISync Resource");
+
+      groupId = parts[0];
+      artifact = parts[1];
+      version = parts[2];
+      classifier = parts[3];
+      packager = parts[4];
+      file = parts[5];
+    }
+
+    public boolean equals(APISyncResource resource, Boolean checkVersion) {
+      if (checkVersion == false) {
+        return groupId.equals(resource.groupId) && artifact.equals(resource.artifact) &&
+            classifier.equals(resource.classifier) && packager.equals(resource.packager) && file.equals(resource.file);
+
+      }
+
+      return groupId.equals(resource.groupId) && artifact.equals(resource.artifact) && version.equals(resource.version)
+          && classifier.equals(resource.classifier) && packager.equals(resource.packager) &&
+          file.equals(resource.file);
+    }
```

Comment:
this can be simplified to 
```
return groupId.equals(resource.groupId) 
    && artifact.equals(resource.artifact) 
    && checkVersion && version.equals(resource.version)
    && classifier.equals(resource.classifier) 
    && packager.equals(resource.packager) 
    && file.equals(resource.file);
```